File: unity/HeightMeasure/Assets/Scripts/Measurement/HeightCalculator.cs
using UnityEngine;

public static class HeightCalculator
{
    // ARKit (via AR Foundation) uses Y-up world space, gravity-aligned by IMU
    public static float VerticalHeight(Vector3 bottom, Vector3 top) => Mathf.Abs(top.y - bottom.y);
    public static float EuclideanDistance(Vector3 a, Vector3 b) => Vector3.Distance(a, b);
}

File: ios/HeightMeasure/Measurement/HeightCalculator.swift
import simd

enum HeightCalculator {
    static func verticalHeight(bottom: SIMD3<Float>, top: SIMD3<Float>) -> Float {
        abs(top.y - bottom.y)
    }

    static func euclideanDistance(a: SIMD3<Float>, b: SIMD3<Float>) -> Float {
        simd_distance(a, b)
    }
}
